contracts/coconsensus/Coconsensus.sol

```diff
@@ -46,23 +46,173 @@ contract Coconsensus is MasterCopyNonUpgradable, GenesisCoconsensus, MosaicVersi
     }
 
 
+    /* Constants */
+
+    /**
+     * Sentinel pointer for marking the ending of circular,
+     * linked-list of genesis metachain ids.
+     */
+    bytes32 public constant SENTINEL_METACHAIN_ID = bytes32(0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff);
+
+
     /* Storage */
 
+    /** Metachain id of the origin chain. */
+    bytes32 public originMetachainId;
+
+    /** Metachain id of the auxiliary chain. */
+    bytes32 public auxiliaryMetachainId;
+
     /** Mapping to track the blocks for each metachain. */
```

Comment:
“Mapping to track the finalised blocks of each metachain”

Reply:
Done.